src/plugins/wizard/public/application/contributions/containers/data_tab/items/use/use_dropbox.tsx

```diff
@@ -118,12 +122,18 @@ export const useDropbox = (props: UseDropboxProps): DropboxProps => {
 
   const onDropField = useCallback(
     (data: FieldDragDataType['value']) => {
-      if (!data) return;
+      if (!data || !validAggTypes.length) {
+        throw new Error('Cannot drop field here, missing drop properties');
+      }
```

Comment:
From a UX perspective, it seems like it would be better to determine the valid drop targets on the drag start event rather than waiting until the drop to declare it invalid. If we did that, we could highlight only valid drop targets, making it clear to the user where that field can go.

Reply:
Agreed. The user will never see this error. It will fail silently and only be logged in the console. The isValidDropTarget property returned by this hook is what actually indicates to the user that the field is droppable. And that is set when the drag begins. I just threw an error here for the console instead of failing silently if the user decided to ignore the UI indicators.

Reply:
Got it, that's helpful context.